operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/cache/PassThroughResourceCache.java

```diff
@@ -0,0 +1,73 @@
+package io.javaoperatorsdk.operator.processing.cache;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.javaoperatorsdk.operator.processing.KubernetesResourceUtils;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PassThroughResourceCache<R extends CustomResource> {
```

Comment:
This should implement `ResourceCache`.

Reply:
Not sure if we want to go that way. If we want to handle caches efficiently, we always need the passthrough cache, and just the underlying store needs to be replaceable, that is why it does not implement ResourceCache.